iterative/resource_task.go

```diff
@@ -369,22 +368,26 @@ func resourceTaskBuild(ctx context.Context, d *schema.ResourceData, m interface{
 		PermissionSet: d.Get("permission_set").(string),
 	}
 
-	name := d.Id()
-	if name == "" {
-		if identifier := d.Get("name").(string); identifier != "" {
-			name = identifier
-		} else if identifier := os.Getenv("GITHUB_RUN_ID"); identifier != "" {
-			name = identifier
-		} else if identifier := os.Getenv("CI_PIPELINE_ID"); identifier != "" {
-			name = identifier
-		} else if identifier := os.Getenv("BITBUCKET_STEP_TRIGGERER_UUID"); identifier != "" {
-			name = identifier
+	id, err := common.ParseIdentifier(d.Id())
+	if err != nil {
+		if name := d.Get("name").(string); name != "" {
+			if newId, err := common.ParseIdentifier(name); err == nil {
+				id = newId
+			} else {
+				id = common.NewIdentifier(name)
+			}
```

Comment:
```suggestion
			id = common.NewIdentifier(name)
			if newId, err := common.ParseIdentifier(name); err == nil {
				id = newId
			}
```

Also... I have seen this code before 🤔

Reply:
Yes, sounds familiar; I'll RM [^1] for now, though. 😅 

[^1]: Repeat myself.